panel/links.py

```diff
@@ -0,0 +1,295 @@
+    def __init__(self, root_model, link, source, target=None):
+        self.root_model = root_model
+        self.link = link
+        self.source = source
+        self.target = target
+        self.validate()
+
+        references = {k: v for k, v in link.get_param_values()
+                      if k not in ('source', 'target', 'name', 'code')}
+
+        src_model = self._resolve_model(
+            root_model, source, self._get_source_spec(link))
+        references['source'] = src_model
+
+        tgt_model = self._resolve_model(
+            root_model, target, self._get_target_spec(link))
+        if tgt_model is not None:
+            references['target'] = tgt_model
+
+        if 'holoviews' in sys.modules:
+            if is_bokeh_element_plot(source):
+                for k, v in source.handles.items():
+                    if isinstance(v, BkModel):
+                        references['source_' + k] = v
+            if is_bokeh_element_plot(target):
+                for k, v in target.handles.items():
+                    if isinstance(v, BkModel):
+                        references['target_' + k] = v
+
+        self._initialize_models(link, src_model, tgt_model)
+        self._process_references(references)
+
+        code = self._get_code(link)
+
+        src_cb = CustomJS(args=references, code=code)
+        changes, events = self._get_triggers(link)
+        for ch in changes:
+            src_model.js_on_change(ch, src_cb)
+        for ev in events:
+            src_model.js_on_event(ev, src_cb)
+
+    def _get_source_spec(self, link):
+        """
+        Returns the source model specification.
+        """
+        return None
+
+    def _get_target_spec(self, link):
+        """
+        Returns the target model specification.
+        """
+        return None
+
+    def _get_code(self, link):
+        """
+        Returns the code to be executed.
+        """
+        return ''
+
+    def _get_triggers(self, link):
+        """
+        Returns the changes and events that trigger the callback.
+        """
+        return [], []
+
+    def _initialize_models(self, link, src_model, tgt_model):
+        """
+        Applies any necessary initialization to the source and target
+        models.
+        """
+        pass
+
+    def validate(self):
+        pass
+
+
+class WidgetLinkCallback(LinkCallback):
+
+    source_code = """
+    target['{value}'] = source.value
+    """
+
+    def _get_target_spec(self, link):
+        return link.target_model
+
+    def _initialize_models(self, link, src_model, tgt_model):
+        if link.target_property and tgt_model:
+            setattr(tgt_model, link.target_property, src_model.value)
+        if tgt_model is None and not link.code:
+            raise ValueError('Model could not be resolved on target '
+                             '%s and no custom code was specified.' % 
+                             type(self.target).__name__)
+
+    def _process_references(self, references):
```

Comment:
Should not a default method be added in the Parent class?

Reply:
Yes, my oversight.